### PR TITLE
Token reintegration

### PR DESCRIPTION
**Pull request by stevex86**
Integrated new Juniper tokens, eliminated old TokenSchema struct (it's not necessary anymore), and added documentation to exported fields that were not documented.

### CODE REVIEW

File: authenticate.go [hunk 1]
```diff
@@ -3,23 +3,37 @@ package friends
...
-	if !token.VerifyToken(tokenString, time.Duration(7*24*time.Hour), []byte("change me when deployed")) {
+	tokenObj, err := token.ParseToken(tokenString)
+
+	if err != nil {
 		http.Error(response, "Unauthorized; invalid token", http.StatusUnauthorized)
 	}
-	c.Map(TokenSchema{Token: tokenString})
+
+	signed, err := tokenObj.IsSignedWithKey([]byte("change me when deployed"))
+	if err != nil || !signed {
```
Comment (tforrest) [inline authenticate.go]: Is there ever a chance of having both of them nil?

File: authenticate.go [hunk 2]
```diff
@@ -3,23 +3,37 @@ package friends
...
 	tokenString := authorizationComponents[1]
-	if !token.VerifyToken(tokenString, time.Duration(7*24*time.Hour), []byte("change me when deployed")) {
+	tokenObj, err := token.ParseToken(tokenString)
+
+	if err != nil {
 		http.Error(response, "Unauthorized; invalid token", http.StatusUnauthorized)
 	}
-	c.Map(TokenSchema{Token: tokenString})
+
+	signed, err := tokenObj.IsSignedWithKey([]byte("change me when deployed"))
```
Comment (nfegard) [inline authenticate.go]: We should entertain making the key part of Juniper so that it isn't a literal within every repo.

File: authenticate.go [hunk 3]
```diff
@@ -3,23 +3,37 @@ package friends
...
 func Authenticate(c martini.Context, ctx context.Context, response http.ResponseWriter, r *http.Request) {
 	authorizationComponents := strings.Split(r.Header.Get("Authorization"), " ")
 	if len(authorizationComponents) < 2 || !strings.EqualFold(authorizationComponents[0], "Bearer") {
 		http.Error(response, "Unauthorized; invalid authorization header", http.StatusUnauthorized)
 	}
 
 	tokenString := authorizationComponents[1]
-	if !token.VerifyToken(tokenString, time.Duration(7*24*time.Hour), []byte("change me when deployed")) {
+	tokenObj, err := token.ParseToken(tokenString)
+
```
Comment (nfegard) [inline authenticate.go]: don't need this new line

File: authenticate.go [hunk 4]
```diff
@@ -3,23 +3,37 @@ package friends
...
+	if err != nil {
 		http.Error(response, "Unauthorized; invalid token", http.StatusUnauthorized)
 	}
-	c.Map(TokenSchema{Token: tokenString})
+
+	signed, err := tokenObj.IsSignedWithKey([]byte("change me when deployed"))
+	if err != nil || !signed {
+		http.Error(response, "Unauthorized; invalid token", http.StatusUnauthorized)
+	} else if tokenObj.IsStale(time.Duration(7 * 24 * time.Hour)) {
```
Comment (nfegard) [inline authenticate.go]: The duration for assessing staleness should be a value kept somewhere accessible so we don't have to type it out and risk discrepancies.

File: service.go [hunk 5]
```diff
@@ -4,37 +4,31 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
-	"strings"
 
 	"golang.org/x/net/context"
 	"google.golang.org/appengine"
 	"google.golang.org/appengine/datastore"
 
+	"github.com/BrambleLLC/juniper/middleware"
```
Comment (nfegard) [inline service.go]: is middleware a temp name?

Comment (stevex86) [inline service.go]: Permanent for as long as it is sensible

File: authenticate.go (hunk 4 above)
Comment (stevex86) [inline authenticate.go]: I've incorporated this feedback here: https://github.com/BrambleLLC/juniper/pull/5

File: authenticate.go (hunk 2 above)
Comment (stevex86) [inline authenticate.go]: I've incorporated this feedback here: https://github.com/BrambleLLC/juniper/pull/5

File: authenticate.go (hunk 3 above)
Comment (stevex86) [inline authenticate.go]: ಠ_ಠ

File: authenticate.go (hunk 2 above)
Comment (stevex86) [inline authenticate.go]: As soon as we merge juniper#5, I will make the necessary changes to use the constants here.

File: authenticate.go (hunk 4 above)
Comment (stevex86) [inline authenticate.go]: As soon as we merge juniper#5, I will make the necessary changes to use the constants here.

Comment (nfegard) [inline authenticate.go]: Merge has been made, please update @stevex86

**Pull request merged**